media/windows/ECC-MQTT-IoT/ECC_MQTT_IoT_SQLite.py

```diff
@@ -101,16 +124,149 @@
     print(F"ECC MQTT IoT Listener application, version {eccmqtt_iot_version}, {eccmqtt_iot_date}...")
     sys.exit(0)
 
+
+class GZipRotator(logging.handlers.RotatingFileHandler):
+
+    def doRollover(self):
+
+        global done_flag
+
+        # Let the default log rollover routine do its thing first...
+        super().doRollover()
+
+        # Construct a file pattern to match all current gzip'd logs
+        base_file_pattern = self.baseFilename + ".[0-9].gz"
+        # Get a list of the current .gzip'd log files to determine the next extension to use
+        arch_files = glob.iglob(base_file_pattern)
+        # Loop through the archived log files to find the last file number in use
+        lastgz_file_num = '0'  # default value for first iteration (no current gzips)
+        for archFile in arch_files:
+            fn = archFile.split('.')
+            # The gzip'd file "number" is in array element 2
+            lastgz_file_num = fn[2]
+        lastgz_file_int = int(lastgz_file_num)
+        # Construct a gzip file name for the destination by incrementing the last file number
+        # used.  If more than (backupCount) files are already zipped, delete the oldest and rename
+        # the remaining to free up the last number for re-use.  This ensures that the newest gzip
+        # will always be the highest number.
+        nextgz_file_int = lastgz_file_int + 1
+        if nextgz_file_int > self.backupCount:
+            nextgz_file_int = self.backupCount
+            # Since we already have the maximum gzip'd logs, delete the oldest, then rename the
+            # remaining to free up the last slot.
+            oldest_gz_log = self.baseFilename + ".1.gz"
+            os.remove(oldest_gz_log)
+            # Rename the remaining gzip'd logs...
+            arch_files = glob.iglob(base_file_pattern)
+            for archFile in arch_files:
+                fn = archFile.split('.')
+                # The gzip'd file "number" is in array element 2
+                newgz_file_num = int(fn[2]) - 1
+                newgz_file = fn[0] + "." + fn[1] + "." + str(newgz_file_num) + ".gz"
+                os.rename(archFile, newgz_file)
```

Comment:
I think this iglob scheme will have the same problem as the prior loop: you can't assume that the list of files you get back will be in sorted, numeric order.

Reply:
Yeah, same comment as above...